gridscale/resource_gridscale_network.go

```diff
@@ -257,35 +257,38 @@ func resourceGridscaleNetworkUpdate(d *schema.ResourceData, meta interface{}) er
 	errorPrefix := fmt.Sprintf("update network (%s) resource -", d.Id())
 
 	labels := convSOStrings(d.Get("labels").(*schema.Set).List())
-	requestBody := gsclient.NetworkUpdateRequest{
+	requestBody := gsclient.NetworkUpdatePutRequest{
 		Name:       d.Get("name").(string),
 		L2Security: d.Get("l2security").(bool),
 		Labels:     &labels,
 	}
-	if d.HasChange("dhcp_active") {
-		dhcpActive := d.Get("dhcp_active").(bool)
-		requestBody.DHCPActive = &dhcpActive
-	}
-	if d.HasChange("dhcp_range") {
-		dhcpRange := d.Get("dhcp_range").(string)
+
+	dhcpActive := d.Get("dhcp_active").(bool)
+	requestBody.DHCPActive = &dhcpActive
+
+	dhcpRange, isSet := d.Get("dhcp_range").(string)
+	if isSet && dhcpRange != "" {
 		requestBody.DHCPRange = &dhcpRange
 	}
-	if d.HasChange("dhcp_gateway") {
-		dhcpGateway := d.Get("dhcp_gateway").(string)
+
+	dhcpGateway, isSet := d.Get("dhcp_gateway").(string)
+	if isSet && dhcpGateway != "" {
```

Comment:
As I user, I've to set the `dhcp_gateway` to None/null in the TF to unset the value. is that correct?